CountryPickeriOS13+/CountryPickerSPM/Sources/CountryPickerSPM/CountryManager.swift

```diff
@@ -0,0 +1,242 @@
+    ///
+    /// - Note: By default, countries can be filtered by their country names
+    internal var filters: Set<CountryFilterOption> = [.countryName]
+    
+    private init() {}
+}
+
+public extension CountryManager {
+    
+    /// Fetch country list from a given property list file path
+    ///
+    /// - Parameter path: URL for the country plist file path
+    /// - Returns: A list of countries sorted by `CountryName`
+    
+    func fetchCountries(fromURLPath path: URL) throws -> [Country] {
+        guard let rawData = try? Data(contentsOf: path),
+              let countryCodes = try? PropertyListSerialization.propertyList(from: rawData, format: nil) as? [String] else {
+            throw "[CountryManager] ❌ Missing countries plist file from path: \(path)"
+        }
+        
+        // Sort country list by `countryName`
+        let sortedCountries = countryCodes.map { Country(countryCode: $0) }.sorted { $0.countryName < $1.countryName }
+        
+        #if DEBUG
+        print("[CountryManager] ✅ Successfully prepared list of \(sortedCountries.count) countries")
+        #endif
+        
+        return sortedCountries
+    }
+    
+    /// Prepares a country list object for usage while clearing any cached countries
+    ///
+    /// - Throws: Incase country list preperation fails to determine or convert data from a given URL file path.
+    func loadCountries() throws {
+        let url = URL(fileURLWithPath: countriesFilePath ?? "")
+        let fetchedCountries = try fetchCountries(fromURLPath: url)
+        countries.removeAll()
+        countries.append(contentsOf: fetchedCountries)
+    }
+    
+    func allCountries(_ favoriteCountriesLocaleIdentifiers: [String]) -> [Country] {
+        favoriteCountriesLocaleIdentifiers
+            .compactMap { country(withCode: $0) } + countries
+    }
+    
+     /// As the function name suggests, resets the last selected country
+    func resetLastSelectedCountry() {
+        lastCountrySelected = nil
+    }
+    
+    func filterCountries(searchText: String) -> [Country] {
+        countries.compactMap { (country) -> Country? in
+            // Filter country by country name first character
+            if filters.contains(.countryName), country.countryName.capitalized.contains(searchText.capitalized) {
+                return country
+            }
+            
+            // Filter country by country code and utilize `CountryFilterOptions`
+            if filters.contains(.countryCode), country.countryCode.capitalized.contains(searchText.capitalized) {
+                return country
+            }
+            
+            // Filter country by digit country code and utilize `CountryFilterOptions`
+            if filters.contains(.countryDialCode), let digitCountryCode = country.digitCountrycode, digitCountryCode.contains(searchText) {
+                return country
+            }
+            return nil
+        }.removeDuplicates()
```

Comment:
How about we use set instead of array and that way we do not have to worry about duplicates.